src/qcodes/instrument/instrument_base.py

```diff
@@ -23,6 +24,8 @@
 
 log = logging.getLogger(__name__)
 
+TParameter = TypeVar("TParameter", bound=ParameterBase)
```

Comment:
We may be able to remove the cast if we use the new default argument to TypeVar (and use typevar from typing_extensions)